internal/indexer/indexer.go

```diff
@@ -142,6 +144,39 @@ func (i *Indexer) loadPackages() error {
 		i.packages = pkgs
 		i.packagesByFile = map[string][]*packages.Package{}
 
+		shouldVisitPackage := func(p *packages.Package) bool {
```

Comment:
Oh cool. Could we extract this into a method on the indexer and add a few unit tests around it?

Reply:
I think this could be useful outside of this PR - it might be worth breaking this out and merging it earlier.

Reply:
I'm looking at doing this as a first PR.